test/NftToken.test.ts

```diff
@@ -0,0 +1,216 @@
+
+            it("should deploy with correct baseURI", async () => {
+                expect(await nftToken.baseURI()).to.equal(BASE_URI);    
+            });
+
+            it("should deploy with correct initial сost", async () => {
+                expect(await nftToken.cost()).to.equal(INITIAL_COST);    
+            });
+
+        });
+
+        describe("Vesting Contract Owner Test Cases 👮", function () {
+
+            after(async function () {
+                await snapshotB.restore();
+            });
+
+            it("should change baseURI", async () => {
+                let newBaseUri = "https://ipfs.io/ipfs/1/"
+
+                const tx = await nftToken.setBaseURI(newBaseUri);
+                await expect(tx).to.emit(
+                    nftToken,
+                  "SetBaseURI"
+                ).withArgs(
+                    newBaseUri
+                );
+
+                expect(await nftToken.baseURI()).to.equal(newBaseUri); 
+            });
+
+            it("shouldn't change baseURI from the non-current owner", async () => {
+                let newBaseUri = "https://ipfs.io/ipfs/1/"
+
+                await expectRevert(
+                    nftToken.connect(user1).setBaseURI(newBaseUri),
+                    "Ownable: caller is not the owner"
+                );
+            });
+
+            it("should change NFT cost", async () => {
+                let newCost = ethers.utils.parseEther("0.02");
+
+                const tx = await nftToken.setCost(newCost);  
+                await expect(tx).to.emit(
+                    nftToken,
+                  "SetCost"
+                ).withArgs(
+                    newCost
+                );
+
+                expect(await nftToken.cost()).to.equal(newCost); 
+            });
+
+            it("shouldn't change baseURI from the non-current owner", async () => {
+                let newCost = ethers.utils.parseEther("0.02");
+
+                await expectRevert(
+                    nftToken.connect(user1).setCost(newCost),
+                    "Ownable: caller is not the owner"
+                );
+            });
+
+        });
+
+        describe("Vesting Contract Buy 💵 and Withdraw 💳 Test Cases", function () {
+
+            let tokenId = BigNumber.from(1);
+
+            after(async function () {
+                await snapshotB.restore();
+            });
+
+            it("should sell NFT to the user", async () => {
+                const tx = await nftToken.connect(user1).buy({value: INITIAL_COST});
+                await expect(tx).to.emit(
+                    nftToken,
+                  "Buy"
+                ).withArgs(
+                    user1.address,
+                    tokenId,
+                    INITIAL_COST
+                );
+
+                expect(await nftToken.balanceOf(user1.address)).to.equal(tokenId); 
+                expect(await nftToken.ownerOf(tokenId)).to.equal(user1.address); 
+                tokenId = tokenId.add(1);
```

Comment:
add tests to check balance changes of user and contract. Use openzeppelin test helpers balance tracker (tracker.deltaWithFees) as tx will cost some gas
[https://docs.openzeppelin.com/test-helpers/0.5/api#tracker](url)